site/topics/numbers/number-bases.rst

```diff
@@ -418,6 +418,301 @@ Hexadecimal (Base 16)
 Converting Numbers Between Bases
 ================================
 
+* Although only three bases were discussed so far, but there is nothing stopping one from using base 5, or 123, or ...
+
+* A subscript is added to the number to specify the base when there is a potential for ambiguity
+* For example, consider the number :math:`10`
+
+    * What number is that?
+    * Is it a base ten number?
+    * Base two?
+    * Hexadecimal?
+
+
+* Ambiguity is eliminated by including the subscript
+
+    * :math:`10_{10}` means ten
+    * :math:`10_{2}` means two
+    * :math:`10_{16}` means sixteen
+
+
+* In a computing context, binary and hexadecimal are often written as ``0b10`` and ``0x10`` respectively
+
+* Notice that, with any base greater than 1, the value of :math:`10` will always be the base
+
+    * Given a base :math:`b`
+    * The second digit is always associated with the value :math:`b^{1}`
+    * And if the value in the second digit is :math:`1`, then the number is :math:`1 \times b^{1} == 1 \times b == b`
+
+
+Converting to Decimal
+---------------------
+
+* It is often useful to convert numbers from other bases to decimal since that is what is used in everyday life
+
+    * Saying :math:`123` is a lot easier for us to understand than saying :math:`1111011_{2}` or :math:`7B_{16}`
+    * However, an encoding is only an encoding --- these all *mean* the same thing
+
+
+* For both the binary and hexadecimal bases discussed, numbers were already converted to decimal
+
+    * :math:`1111011_{2}`
+
+        :math:`(1 \times 2^{6} + 1 \times 2^{5} + 1 \times 2^{4} + 1 \times 2^{3} + 0 \times 2^{2} + 1 \times 2^{1} + 1 \times 2^{0})_{10}`
+
+        :math:`(1 \times 64 + 1 \times 32 + 1 \times 16 + 1 \times 8 + 0 \times 4 + 1 \times 2 + 1 \times 1)_{10}`
+
+        :math:`(64 + 32 + 16 + 8 + 0 + 2 + 1)_{10}`
+
+        :math:`123_{10}`
+
+
+    * :math:`7B_{16}`
+
+        :math:`(7 \times 16^{1} + 11 \times 16^{0})_{10}`
+
+        :math:`(7 \times 16 + 11 \times 1)_{10}`
+
+        :math:`(112 + 11)_{10}`
+
+        :math:`123_{10}`
+
+
+* In general, multiply the value of the digit in base ten by the value of the base in base ten to the power of its corresponding position
+
+* For example, given some number of :math:`n` digits :math:`d` in base :math:`b`
+
+    :math:`(d_{n-1}d_{n-2}...d_{2}d_{1}d_{0})_{b}`
+
+
+* The value in base ten would be calculated as follows
+
+    :math:`(d_{n-1} \times b^{n-1} + d_{n-2} \times b^{n-2} + ... + d_{2} \times b^{2} + d_{1} \times b^{1} + d_{0} \times b^{0})_{10}`
+
+
+* Consider the number :math:`J204_{20}` --- note that :math:`J_{20} == 19_{10}` in this example
+
+
+    :math:`(19 \times 20^{3} + 2 \times 20^{2} + 0 \times 20^{1} + 4 \times 20^{0})_{10}`
```

Comment:
![image](https://github.com/jameshughes89/cs2XX-ComputerArchitecture/assets/5175174/b253a7a8-7d0d-4359-aaad-d91c50a25478)

Google kindly even let me know that I should have been looking for the 10th letter, because of 1-based indexing.

Reply:
I'm unclear on what this means. Or are you pointing out the amusingness of Google?
```
0 0
1 1
2 2 
3 3
4 4
5 5
6 6
7 7
8 8
9 9
10 A
11 B
12 C
13 D
14 E
15 F
16 G
17 H
18 I
19 J
```

Reply:
I didn't want to count alphabet letters to confirm J was 19.